packages/notifications/doc/migration.md

```diff
@@ -0,0 +1,68 @@
+# Migration
+
+## V5 to V6 migration
+
+In version 6, the notifications package was decoupled from redux. The package now uses only React api to manage its state.
+
+### Using notifications middleware and redux store
+
+Replace `NotificationsPortal` component with `NotificationsProvider`
+
+```diff
+- <NotificationsPortal />
++ <NotificationsProvider />
+# Provider can also render children
+- <NotificationsPortal />
+- {children}
++ <NotificationsProvider>
++  {children}
++ </NotificationsProvider>
+```
+
+Remove the notifications middleware from reducer from your store
+
+```diff
+- import notificationsMiddleware from '@redhat-cloud-services/frontend-components-notifications/notificationsMiddleware';
+- import { notificationsReducer } from '@redhat-cloud-services/frontend-components-notifications/redux';
+
+const middlewares = [
+  thunk,
+  promiseMiddleware,
+-  notificationsMiddleware({
+-    errorTitleKey: ['statusText', 'message', 'errors[0].status'],
+-    errorDescriptionKey: ['errors[0].detail', 'errors', 'stack'],
+-  }),
+  reduxLogger,
+].filter((middleware) => typeof middleware === 'function');
+```
+
+#### Notifications middleware
+
+In v6, you can no longer dispatch automated notifications via redux middleware in combination with promise middleware. If you still need this functionality, you will have to implement your own middleware. You can take inspiration from the legacy version.
```

Comment:
Do we know if people were actually use the middleware?

Reply:
Oh yes. This was a primary method for a long time.